src/DurableTask.Netherite/OrchestrationService/OrchestrationMessageBatch.cs

```diff
@@ -162,20 +168,36 @@ public void ConfirmDurable(Event evt)
             this.workItem.Partition.EnqueueOrchestrationWorkItem(this.workItem);
         }
 
-        bool IsExecutableInstance(TaskOrchestrationWorkItem workItem, out string message)
+        bool IsExecutableInstance(TaskOrchestrationWorkItem workItem, out string message, out bool traceWarningEvenForTimers)
         {
+            message = default;
+            traceWarningEvenForTimers = default;
+
             if (workItem.OrchestrationRuntimeState.ExecutionStartedEvent == null
                 && !this.MessagesToProcess.Any(msg => msg.Event is ExecutionStartedEvent))
             {
                 if (DurableTask.Core.Common.Entities.AutoStart(this.InstanceId, this.MessagesToProcess))
                 {
-                    message = default;
                     return true;
                 }
                 else
                 {
-                    message = workItem.NewMessages.Count == 0 ? "No such instance" : "Instance is corrupted";
-                    return false;
+                    if (DurableTask.Core.Common.Entities.IsEntityInstance(this.InstanceId))
+                    {
+                        message = "Instance is an entity that cannot process this type of message";
+                        traceWarningEvenForTimers = true;
+                    }
+                    else if (workItem.OrchestrationRuntimeState.Events.Count == 0)
+                    {
+                        message = "Instance does not exist (it may have been purged)";
+                        traceWarningEvenForTimers = false;
+                    }
+                    else
+                    {
+                        message = $"Instance has an invalid history ({workItem.OrchestrationRuntimeState.Events} events)";
+                        traceWarningEvenForTimers = true;
```

Comment:
I'm not immediately seeing the relationship between these if-else cases, and the different values for `traceWarningEvenForTimers`. Can you please clarify them?

Reply:
I can add comments. The point is that some of these warnings are common place when using timers. I have tests where timers are canceled, the orchestration completes, and then the orchestration is purged. But when the timer still fires later, we get a message for a non-existent orchestration. Because this is completely within the normal operation and use of orchestrations, I do not want to issue a warning in this case.